src/components/ui4/ButtonUi4.tsx

```diff
@@ -114,41 +113,56 @@ export function ButtonUi4(props: Props) {
 
   const { spinnerColor, textStyle, gradientProps } = buttonProps[type]
 
-  const dynamicGradientStyles = {
-    alignSelf,
-    opacity: disabled ? 0.3 : pending ? 0.7 : 1
-  }
-
   // Show a spinner if waiting on the onPress promise OR if the spinner prop is
   // manually enabled.
   const hideContent = pending || spinner
 
+  const dynamicGradientStyles = React.useMemo(
+    () => ({
+      alignSelf,
+      opacity: disabled ? 0.3 : hideContent ? 0.7 : 1
+    }),
+    [alignSelf, disabled, hideContent]
+  )
+
+  // Hide the content by setting its opacity to 0 instead of removing it. This
+  // will allow the button to remain constant in size.
+  const contentOpacityStyle = React.useMemo(() => ({ opacity: hideContent ? 0 : 1 }), [hideContent])
+
   const maybeText =
     label == null ? null : (
       <EdgeText numberOfLines={1} disableFontScaling style={textStyle}>
         {label}
       </EdgeText>
     )
 
-  const containerStyle: ViewStyle[] = [styles.containerCommon]
-  if (layout === 'column') containerStyle.push(styles.containerColumn)
-  if (layout === 'row') containerStyle.push(styles.containerRow)
-  if (layout === 'solo') containerStyle.push(styles.containerSolo)
+  const containerStyle = React.useMemo(() => {
+    const retStyle: ViewStyle[] = [styles.containerCommon]
+    if (layout === 'column') retStyle.push(styles.containerColumn)
+    if (layout === 'row') retStyle.push(styles.containerRow)
+    if (layout === 'solo') retStyle.push(styles.containerSolo)
 
-  if (type === 'tertiary') containerStyle.push(styles.containerTertiary)
+    if (type === 'tertiary') retStyle.push(styles.containerTertiary)
+    return retStyle
+  }, [layout, styles.containerColumn, styles.containerCommon, styles.containerRow, styles.containerSolo, styles.containerTertiary, type])
 
   const customMargin = marginRem == null ? undefined : sidesToMargin(mapSides(fixSides(marginRem, 0), theme.rem))
   const customPadding = paddingRem == null ? undefined : sidesToPadding(mapSides(fixSides(paddingRem, 0), theme.rem))
-  const finalContainerCommon = [styles.containerCommon, containerStyle, customMargin, customPadding]
+  const finalContainerCommon = React.useMemo(
+    () => [styles.containerCommon, containerStyle, customMargin, customPadding],
+    [containerStyle, customMargin, customPadding, styles.containerCommon]
+  )
 
   return (
-    <TouchableOpacity disabled={disabled || pending} style={finalContainerCommon} onPress={handlePress} testID={testID}>
+    <TouchableOpacity disabled={disabled || pending || spinner} style={finalContainerCommon} onPress={handlePress} testID={testID}>
       <LinearGradient
         {...gradientProps}
         style={[styles.contentCommon, dynamicGradientStyles, mini ? styles.contentSizeMini : styles.contentSizeDefault, ...finalContainerCommon]}
       >
-        {hideContent ? null : children}
-        {hideContent ? null : maybeText}
+        <View style={contentOpacityStyle}>
```

Comment:
This is a dangerous change! This is the type of high regression potential change I was referring to this week.

You are changing the way children are laid out in the button. This needs a full audit of everywhere ButtonUi4 is used. Note the difference before and after your change in the screenshots
<img width="374" alt="Screenshot 2024-01-27 at 17 27 08" src="https://github.com/EdgeApp/edge-react-gui/assets/6079354/e6a62bda-0366-4bc0-b497-e80af2b639d9">

<img width="380" alt="Screenshot 2024-01-27 at 17 27 22" src="https://github.com/EdgeApp/edge-react-gui/assets/6079354/aeca24fc-57e0-4658-880f-366cc70b98da">

The first image is not ideal but the issue is that it's different which it shouldn't be.

Change the View alignment style that affects children to exactly match what the LinearGradient had. Or better yet, punt this to 4.1 as it's not that urgent.